drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -463,13 +463,80 @@ static int siwx91x_scan(const struct device *dev, struct wifi_scan_params *z_sca
 static int siwx91x_status(const struct device *dev, struct wifi_iface_status *status)
 {
 	struct siwx91x_dev *sidev = dev->data;
+	sl_wifi_interface_t interface;
+	sl_si91x_rsp_wireless_info_t wlan_info = { };
+	int ret;
 	int32_t rssi = -1;
 
+	__ASSERT(status, "status cannot be NULL");
+
 	memset(status, 0, sizeof(*status));
 	status->state = sidev->state;
-	sl_wifi_get_signal_strength(SL_WIFI_CLIENT_INTERFACE, &rssi);
-	status->rssi = rssi;
-	return 0;
+
+	interface = sl_wifi_get_default_interface();
+	ret = sl_wifi_get_wireless_info(&wlan_info);
+	if (ret) {
+		LOG_ERR("Failed to get the wireless info: 0x%x", ret);
+		return -EIO;
+	}
+
+	strncpy(status->ssid, wlan_info.ssid, WIFI_SSID_MAX_LEN);
+	status->ssid_len = strlen(status->ssid);
+	memcpy(status->bssid, wlan_info.mac_address, WIFI_MAC_ADDR_LEN);
+	status->mfp = WIFI_MFP_REQUIRED;
+
+	if (FIELD_GET(SL_WIFI_2_4GHZ_INTERFACE, interface)) {
```

Comment:
`SL_WIFI_2_4GHZ_INTERFACE` is not a mask. I would either write:
   - `FIELD_GET(SIWX91X_BAND_MASK, interface) == 1`
   - `interface & SL_WIFI_2_4GHZ_INTERFACE`